reference_guide/settings_guide.md

```diff
@@ -0,0 +1,154 @@
+* Application Settings implementations, declared using the [`applicationConfiguration` EP](#ep-for-declaring-application-settings), must have a default constructor with no arguments. 
+* Project Settings implementations, declared using the [`projectSettings` EP](#ep-for-declaring-project-settings), must declare a constructor with a single argument of type [`Project`](upsource:///platform/core-api/src/com/intellij/openapi/project/Project.java).
+
+For a `Configurable` implementation properly declared using an EP, the implementation's constructor is not called by the IntelliJ Platform until the corresponding Settings `displayName` entry is chosen in the Settings Dialog menu.
+For performance reasons, do not create any Swing components in a constructor.
+
+> **Warning** The IntelliJ Platform may instantiate a `Configurable` implementation on a background thread, so creating Swing components in a constructor can degrade UI responsiveness.  
+
+#### IntelliJ Platform Interactions with Configurable
+Interactions between the IntelliJ Platform and a generic implementation of the `Configurable` interface are documented in the source file.
+A few high points are reviewed here:
+* After calling `Configurable.createComponent()`, the IntelliJ Platform immediately calls `Configurable.reset()` to initialize the values of the Settings.
+  So initialization of Setting values in the constructor or `createComponent()` is unnecessary.
+* See the [Constructors](#constructors) section for information about when a Settings object is instantiated.
+* Once instantiated, a `Configurable` instance is valid until the user presses **OK** or **Cancel** in the Settings Dialog.
+  This lifetime holds true regardless of whether the implementation's Settings are changed, or the user chooses a different entry on the Settings Dialog menu. 
+  The IntelliJ Platform calls the `Configurable.disposeUIResources()` method as the last call to an instantiated Settings implementation.
+
+#### Configurable Marker Interfaces
+Implementations based on `Configurable` can have nested interfaces, which provide additional flexibility in the implementation.
+See the `Configurable` interface file for details.
+ 
+The following nested interfaces are markers, which convey information about the Swing form to the IntelliJ Platform:
+  * `Configurable.NoScroll` - Notifies the Settings dialog not to add scroll bars to the Swing form.
+    By default, a plugin's Settings component is put into a scrollable pane.
+    However, a Settings panel can have a JTree, which requires its own JScrollPane. So NoScroll interface should be used to remove outer JScrollPane.
+  * `Configurable.NoMargin` - Notifies the Settings dialog not to add an empty border to the Swing form.
+    By default, an empty border is added for a plugin's Settings component.
+
+#### Additional Interfaces Based on Configurable
+There are over 90 interfaces and classes based on `Configurable` in the IntelliJ Platform code base.
```

Comment:
what 90? how did you get these? can we list more that might be relevant to plugin authors